packages/markers/src/browser/problem/problem-widget.tsx

```diff
@@ -44,6 +44,12 @@ export class ProblemWidget extends TreeWidget {
         this.addClass('theia-marker-container');
 
         this.addClipboardListener(this.node, 'copy', e => this.handleCopy(e));
+        this.toDispose.push(this.model.onSelectionChanged(selection => {
+            const node = selection[0];
```

Comment:
this event is fired even if someone programmatically select a node, you don't want to focus in this case

Reply:
I was following other designs from other tree-widgets which used the same event but I'll try using the suggestion you made.

Reply:
Could you point me to this code?

Reply:
Sure, there's a few other widgets that did it similarly to my initial implementation:

https://github.com/theia-ide/theia/blob/d4adca224ff8e0314101ca792bf98967e4e33184/packages/callhierarchy/src/browser/callhierarchy-tree/callhierarchy-tree-widget.tsx#L53-L57

https://github.com/theia-ide/theia/blob/d4adca224ff8e0314101ca792bf98967e4e33184/packages/typehierarchy/src/browser/tree/typehierarchy-tree-widget.tsx#L48-L53

Reply:
It will bite there if someone trigers programmatic selection. I am not sure that a user then wants to have some editors pop up.

Reply:
> It will bite there if someone trigers programmatic selection. I am not sure that a user then wants to have some editors pop up.

Should we file issues to track and fix them?

Reply:
Let's wait for someone to open. It is probably very rare case.